pages/index.js

```diff
@@ -10,22 +10,66 @@ const Subline = styled.h2`
   cursor: default;
 `;
 
+const initialFilter = {
+  time: null,
+  difficulty: null,
+  category: null,
+  priceCategory: null,
+};
+
 export default function HomePage({ projects, favorites, onToggleFavorite }) {
   const [projectsToDisplay, setProjectsToDisplay] = useState(projects);
+  const [filter, setFilter] = useState(initialFilter);
 
-  function handleFilter(filterData) {
-    setProjectsToDisplay(
-      projects.filter(
-        (project) =>
-          project.priceCategory === filterData.priceCategory &&
-          project.difficulty === filterData.difficulty &&
-          filterData.category.includes(project.category.toLowerCase()) &&
-          filterData.time >= project.time
-      )
-    );
+  useEffect(() => {
+    handleFilter(filter);
+  }, [filter]);
```

Comment:
you don't need a useEffect here. You can directly derive the filtered projects below the states. SInce an update of the "filters" causes the whole function to rerender, the filtering is then executed automatically. And please change the name to plural, since the state holds multiple filters:
```js
const [filters, setFilters] = useState(initialFilter);
const filteredProjects = handleFilters(filters);

// use filteredProjects here instead of "projects"
```